jobs/ServiceInstanceUpdateJob.js

```diff
@@ -51,9 +51,20 @@ class ServiceInstanceUpdateJob extends BaseJob {
         })
         .catch(errors.NotFound, () => undefined)
         .then(resource => _.get(resource, 'spec.options'))
-        .then(resourceDetails => (resourceDetails === undefined ?
-          this.handleInstanceDeletion(instanceDetails, operationResponse, done) :
-          this.updateInstanceIfOutdated(instanceDetails, resourceDetails, operationResponse)))
+        .then(resourceDetails => {
+          if (resourceDetails) {
+            return this.isInstanceUpdateDisabled(resourceDetails)
+              .then(disabled => {
+                if (!disabled) {
+                  return this.updateInstanceIfOutdated(instanceDetails, resourceDetails, operationResponse);
+                }
+                operationResponse.update_init = 'disabled';
```

Comment:
@vinaybheri : can you pls check once we disable scheduled update for any service type can it have update on the scheduled update IT test which runs during deploy ?

@pritishmishra - maybe you can check once with Vinay and make necessary adjustments to that script too to handle this case.

Reply:
Correct me if I am wrong here. But if the Scheduled Update IT triggers an update on the service having the feature disabled, it should anyway succeed. Will that still fail the test?

Reply:
@kamath-prasad -- Its fine for the tests to fail right? We would have disabled it for some reason and we can fail them. Or alternatively if you want to ensure your pipelines are green, IT Test before running the schedule update/backup tests can get this config and proceed with test only if its not disabled.

Reply:
As discussed with you there are 2 things : 

- If some service disables the scheduled update , then deploy-service-fabrik will always fail in the scheduled update test for that service until it gets re-enabled. Hence this can block the whole pipeline since we can never be sure how long that service takes to fix its updates.

- If some update of the test instances fails in post deploy of service-fabrik , then we must auto disable the scheduled update for that service so that it prevents other productive instances from getting updated and hence fail.

So I propose the following flow changes in the scheduled update test in SF post deploy with this PR : 

- Before scheduling , check if the config for that service is disabled . If disabled , then skip the test for that service.
- If some test fails , then update the config to disabled via script for that service. This will ensure that no other instances of that service type will be updated.
- if some test passes , then update the config to enabled via script for that service type.

Reply:
@pritishmishra : hope these proposed changes are taken care in SFPC